src/main/java/org/spongepowered/api/service/economy/Account.java

```diff
@@ -0,0 +1,178 @@
+	 *         account holds.
+	 */
+	Map<Currency, BigDecimal> getBalances();
+
+	/**
+	 * Sets the balance stored by this {@link Account} to the given amount for
+	 * the {@link Currency} specified.
+	 * 
+	 * <p>Negative balances may or may not be supported depending on
+	 * the {@link Currency} specified. See {@link Currency#canGoNegative()}.</p>
+	 * 
+	 * @param currency the {@link Currency} to be used.
+	 * @param amount a {@link BigDecimal} value representing the new balance
+	 *        for the {@link Currency} given.
+	 * @param a {@link Currency} 
+	 */
+	TransactionResult setBalance( Currency currency, BigDecimal amount);
+
+	/**
+	 * Adds the given amount of the specified {@link Currency} to this {@link Account}.
+	 * 
+	 * @param currency the {@link Currency} to be used.
+	 * @param amount a {@link BigDecimal} representative of the amount to give to the target
+	 *        Account
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult deposit(Currency currency, BigDecimal amount);
+	
+	/**
+	 * Adds the given amount of the {@link Currency} to this {@link Account} 
+	 * in the given contexts.
+	 * 
+	 * <p>If the account is not in the given {@link Context} the result should be
+	 * a failure</p>
+	 * 
+	 * @param amount a Number representative of the amount to give to the target
+	 *        Account
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult deposit(Currency currency, Number amount, Set<Context> contexts);
+
+	/**
+	 * Removes the given amount of the {@link Currency} from this {@link Account}.
+	 * 
+	 * @param amount a Number representing the amount to remove from the account
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult withdraw(Currency currency, Number amount);
+	
+	/**
+	 * Removes the given amount of the {@link Currency} from this {@link Account}
+	 * in the given contexts.
+	 * 
+	 * <p>If the account is not in the given {@link Context} the result should be
+	 * a failure</p>
+	 * 
+	 * @param amount a Number representing the amount to remove from the account
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult withdraw(Currency currency, Number amount, Set<Context> contexts);
+
+	/**
+	 * Transfers the given amount of currency from the source {@link Account} to
+	 * the destination Account.
+	 * 
+	 * @param source the Account to take from
+	 * @param destination the Account to send to
+	 * @param amount the amount of currency to transfer
+	 * 
+	 * @return a {@link TransactionResult} representative of the effects of the
+	 *         operation
+	 */
+	TransactionResult transfer(Account to, BigDecimal amount, Currency currency);
```

Comment:
I don't know if this is the right place for this method. I would move it to the EcoService.